pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -203,7 +213,29 @@ func init() {
 }
 
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
-func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {}
+func (vs *VSphere) Initialize(clientBuilder controller.ControllerClientBuilder) {
+	if vs.cfg == nil {
+		return
+	}
+
+	// Only on controller node it is required to register listeners.
+	var err error
+	vs.clientBuilder = clientBuilder
+	vs.client, err = clientBuilder.Client("vSphere-cloud-provider")
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to create kubeclient: %v", err))
+	}
+
+	// Register callbacks for node updates
+	factory := informers.NewSharedInformerFactory(vs.client, 5*time.Minute)
+	vs.nodeInformer = factory.Core().V1().Nodes()
+	vs.nodeInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    vs.NodeAdded,
+		DeleteFunc: vs.NodeDeleted,
+	})
+	go vs.nodeInformer.Informer().Run(wait.NeverStop)
```

Comment:
What happens if the API server temporarily goes down for few mins? Does the node informer automatically connect when it comes back? Could you please test this scenario?

Reply:
Can you please comment on this since handling this scenario is important?